### PR TITLE
Add debug logging for protocol messages

### PR DESCRIPTION
**Pull request by noffle**

### CODE REVIEW

**Comment by noffle** [comment]
Travis is failing because `debug` uses `let` statements, which aren't in Node 4. However, it's passing on Node 6.

File: feed.js [hunk 1]
```diff
@@ -172,6 +183,17 @@ Feed.prototype._onmessage = function (type, data, start, end) {
 }
 
 Feed.prototype._emit = function (type, message) {
+  switch (type) {
```
Comment (mafintosh) [inline feed.js]: can you guard this switch with a debug.enabled or similar? this function is extremely hot

**Comment by noffle** [comment]
added the if clause

**Comment by noffle** [comment]
this is going to have to be rewritten /w the major release just made

**Comment by mafintosh** [comment]
@noffle i can still patch release this on 6 if you need it?

**Comment by noffle** [comment]
@mafintosh Could you patch it into 6? I don't know when we'll be upgrading the mapeo stack to latest hypercore, so this could still be helpful to us & maybe other folx. Thanks!

**Comment by noffle** [comment]
@mafintosh It'd still be helpful to see this patched into 6 :blue_heart:

**Pull request closed without merging**